test/fixtures/other/expression-multiple-roots-stress/input.astro

```diff
@@ -0,0 +1,27 @@
+{
+	Promise.all([
+	[].map((something) => {
+		return [
+			<div></div>,
+			<div></div>,
+			<div></div>,
+			<div></div>
+		]
+	}),
+	[].map((something) => {
+		return <div></div><div></div><div></div><div></div>
+	}),
+	[].map((something) => {
+		return <div></div><div></div><div></div><div></div> + <div></div><div></div><div></div><div></div>
```

Comment:
VS Code's JSX syntax highlighting does not highlight this properly, unrelated to us!